x-pack/legacy/plugins/code/server/queue/delete_worker.ts

```diff
@@ -41,6 +41,17 @@ export class DeleteWorker extends AbstractWorker {
   public async executeJob(job: Job) {
     const { uri } = job.payload;
 
+    try {
+      await this.objectClient.getRepository(uri);
+    } catch (error) {
+      this.log.debug(`Repository ${uri} might not exist. Skip delete job.`);
+      return {
+        uri,
+        // Because the repository does not exist, we can regard the delete job to be successful.
+        res: true,
```

Comment:
probably better to log it `error` and return an error

Reply:
While it'd require a whitelist change, what happens if I theoretically serve a git repo at `myhost.com/../../../../../whatever/i/want/to/delete`, clone it then delete it? I think the repo would exist and pass this check, then.

Reply:
As I mentioned in the email, in reality, it won't happen with this fix because if you clone the repo with such URL, i.e. via API, it will be sanitized and normalized to a different format (<host>/<org>/<name>). So, 1) if you delete via API, then it will only delete the repo in the Code data folder, 2) if you try to hack by posting a delete queue job into the Elasticsearch directly as you mentioned previously, it won't find a valid repo via the exact repo path, and thus won't delete the folder again.

In the meantime, I agree that it's always safe to add more checks over the path. So any concrete suggestions on to what degree the path should be checked against based on your experience?

Reply:
`objectClient.getRepository` just checks ES if there's an index with a derived from the repo name. I could create that, then index the deletion job with `../`s in them, maybe?

I'd encourage always checking if the paths are based out of where you expect them to be, and to flat out err if they contain `../` as there's no legitimate use of these when interacting with Code. Don't normalise and check, just fail completely.

Reply:
Add the `../` check both for the `delete_worker` and also right after when the git url is submitted to code server for clone.